### PR TITLE
Examples for Time built in functions

### PR DESCRIPTION
**Pull request by kksinghinfy04**
Signed-off-by kksinghinfy04 <kamlesh.singh@infosys.com>

### CODE REVIEW

**Review by anderseknert — COMMENTED** [review]
Hey! And thanks for submitting this 😃 I think there's a bit much going on here, but perhaps we can reduce it to a simpler example, using only the time functions without relying on external input?

File: docs/content/policy-reference.md [hunk 1]
```diff
@@ -824,6 +824,48 @@ Timezones can be specified as
...
+    image := input.Name
+    createdns := time.parse_rfc3339_ns(input.Created)
+    timediff := time.diff(createdns, nowns)
+    timediff[0] > 0
+}
+violation[image]{
+    image := input.Name
+    createdns := time.parse_rfc3339_ns(input.Created)
+    timediff := time.diff(createdns, nowns)
+    timediff[0] > 2
```
[the remark this answers is not in the document]
Comment (kksinghinfy04) [reply docs/content/policy-reference.md:854]: My bad, I was supposed to put it like this. Will change create a new PR.

first violation if image > 1 yr and second violation if image > 0 yr 2 months

violation[image]{
    image := input.Name
    createdns := time.parse_rfc3339_ns(input.Created)
    timediff := time.diff(createdns, nowns)
    timediff[0] > 0
}
violation[image]{
    image := input.Name
    createdns := time.parse_rfc3339_ns(input.Created)
    timediff := time.diff(createdns, nowns)
    timediff[0] = 0
    timediff[1] > 2
}

File: docs/content/policy-reference.md [hunk 2]
```diff
@@ -824,6 +824,48 @@ Timezones can be specified as
...
+
+This package will check if a container image is older than 2 months. We will run skopeo inspect for target image and stream the output to OPA. If the image creation time is older than 2 months it results output as `"unsafe"` else the output is `"safe"`.
+
+**time_example.rego**
+```live:example/using_time_builtin:module:openable,readonly
+package time_example
+
+nowns := time.now_ns()
+
+result = "safe"{
```
[the remark this answers is not in the document]
Comment (kksinghinfy04) [reply docs/content/policy-reference.md:837]: Thanks for suggestion, will update and create PR.

File: docs/content/policy-reference.md [hunk 3]
```diff
@@ -824,6 +824,48 @@ Timezones can be specified as
 
 Note that the opa executable will need access to the timezone files in the environment it is running in (see the [Go time.LoadLocation()](https://golang.org/pkg/time/#LoadLocation) documentation for more information).
 
+Below is an example that uses some of the Time built-in functions.
+
+This package will check if a container image is older than 2 months. We will run skopeo inspect for target image and stream the output to OPA. If the image creation time is older than 2 months it results output as `"unsafe"` else the output is `"safe"`.
```
[the remark this answers is not in the document]
Comment (kksinghinfy04) [reply docs/content/policy-reference.md:829]: Thanks for suggestion. Will remove skopeo and rewrite with data input as json.

File: docs/content/policy-reference.md [hunk 4]
```diff
@@ -824,6 +824,48 @@ Timezones can be specified as
 
 Note that the opa executable will need access to the timezone files in the environment it is running in (see the [Go time.LoadLocation()](https://golang.org/pkg/time/#LoadLocation) documentation for more information).
 
+Below is an example that uses some of the Time built-in functions.
+
+This package will check if a container image is older than 2 months.
```
Comment (anderseknert) [inline docs/content/policy-reference.md:829]: It also checks if the image is older than 1 year, right? I'm not sure I understand why you'd want both conditions.

File: docs/content/policy-reference.md [hunk 5]
```diff
@@ -824,6 +824,48 @@ Timezones can be specified as
 
 Note that the opa executable will need access to the timezone files in the environment it is running in (see the [Go time.LoadLocation()](https://golang.org/pkg/time/#LoadLocation) documentation for more information).
 
+Below is an example that uses some of the Time built-in functions.
+
+This package will check if a container image is older than 2 months. 
+
+**time_example.rego**
+```live:example/time:module:openable,merge_down
+package time_example
```
Comment (anderseknert) [inline docs/content/policy-reference.md:833]: I think this policy could probably be reduced to something like:

```rego
package time_example

import future.keywords.in

violation[image.name] {
    some image in input.images
    [year, month, _, _, _, _] := time.diff(time.parse_rfc3339_ns(image.created), time.now_ns())
    outdated(year, month)
}

outdated(year, month) {
    year > 0
}

outdated(year, month) {
    year == 0
    month > 2
}
```

Comment (anderseknert) [reply docs/content/policy-reference.md:833]: I'm wondering though if it wouldn't be better to have a policy demonstrate at least 5-6 of the built-in time functions. Right now the focus is more on the data than the actual time built-ins... WDYT?

Comment (kksinghinfy04) [reply docs/content/policy-reference.md:833]: Thank You for review ! I will work on it.

**Comment by srenatus** [comment]
Sorry for the nagging -- is this still being worked on? 👀

**Comment by srenatus** [comment]
Please don't hesitate re-opening this or another PR when you get a chance to come back to this!

**Pull request closed without merging**